src/test/lib/ApiTest.cpp

```diff
@@ -143,86 +141,176 @@ void QuicTestValidateConfiguration()
+    }
 
     //
     // 256-byte ALPN.
     //
-    TEST_QUIC_STATUS(
-        QUIC_STATUS_INVALID_PARAMETER,
-        MsQuic->ConfigurationOpen(
-            Registration,
-            &TooLongAlpn,
-            1,
-            nullptr,
-            0,
-            nullptr,
-            &LocalConfiguration));
+    {
+        ConfigurationScope LocalConfiguration;
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->ConfigurationOpen(
+                Registration,
+                &TooLongAlpn,
+                1,
+                nullptr,
+                0,
+                nullptr,
+                &LocalConfiguration.Handle));
+    }
 
     //
     // Multiple ALPNs
     //
-    const QUIC_BUFFER TwoAlpns[] = {
-        { sizeof("alpn1") - 1, (uint8_t*)"alpn1" },
-        { sizeof("alpn2") - 1, (uint8_t*)"alpn2" }
-    };
-    TEST_QUIC_SUCCEEDED(
-        MsQuic->ConfigurationOpen(
-            Registration,
-            TwoAlpns,
-            2,
-            nullptr,
-            0,
-            nullptr,
-            &LocalConfiguration));
+    {
+        ConfigurationScope LocalConfiguration;
+        const QUIC_BUFFER TwoAlpns[] = {
+            { sizeof("alpn1") - 1, (uint8_t*)"alpn1" },
+            { sizeof("alpn2") - 1, (uint8_t*)"alpn2" }
+        };
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConfigurationOpen(
+                Registration,
+                TwoAlpns,
+                2,
+                nullptr,
+                0,
+                nullptr,
+                &LocalConfiguration.Handle));
+    }
+
+    //
+    // ConfigurationLoad
+    //
+    {
+        ConfigurationScope LocalConfiguration;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConfigurationOpen(
+                Registration,
+                &GoodAlpn,
+                1,
+                nullptr,
+                0,
+                nullptr,
+                &LocalConfiguration.Handle));
+
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConfigurationLoadCredential(
+                LocalConfiguration,
+                &SelfSignedCredConfig));
+    }
+
+    //
+    // Set Ticket Key (single)
+    //
+    {
+        ConfigurationScope LocalConfiguration;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConfigurationOpen(
+                Registration,
+                &GoodAlpn,
+                1,
+                nullptr,
+                0,
+                nullptr,
+                &LocalConfiguration.Handle));
 
-    MsQuic->ConfigurationClose(
-        LocalConfiguration);
-    LocalConfiguration = nullptr;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConfigurationLoadCredential(
+                LocalConfiguration,
+                &SelfSignedCredConfig));
+
+        QUIC_TICKET_KEY_CONFIG KeyConfig;
+        CxPlatZeroMemory(&KeyConfig, sizeof(KeyConfig));
+        KeyConfig.MaterialLength = 64;
+        TEST_QUIC_SUCCEEDED_OR_NOT_SUPPORTED(
```

Comment:
Won't this mask errors when support is added for other TLS?

Reply:
Only NOT_SUPPORTED. I essentially want to say NOT_SUPPORTED is an acceptable return from here; essentially treating the feature as optional as far as the tests are concerned.